### PR TITLE
Consolidate props `blockedDates`, `unblockedDates`, `blockedByDefault` into prop `isDayBlocked`

### PR DESCRIPTION
**Pull request by Corei13**

### CODE REVIEW

**Comment by majapw** [comment]
I think it's a great idea to accept a function for blockedDates but (1) I think it's maybe a bit problematic to overload the prop in this way and (2) having a method kind of makes the `blockedDates`, `unblockedDates`, `blockedByDefault` props unnecessary as you can just pass in `(day) => blockedDatesArray.contains(day)` or `(day) => !blockedDatesArray.contains(day)`. 

It might be better to consolidate this all into one prop, a la `isDayBlocked: PropTypes.func`. What do you think?

**Comment by Corei13** [comment]
@majapw Agreed, that was my first intention actually. But I didn't want to make any breaking changes. I'll go ahead and consolidate the props into one.

File: stories/DateRangePicker.js [hunk 1]
```diff
@@ -83,12 +83,11 @@ storiesOf('DateRangePicker', module)
   ))
   .add('with some blocked dates', () => (
     <DateRangePickerWrapper
-      blockedDates={datesList}
+      isDayBlocked={day1 => datesList.some(day2 => day2.isSame(day1, 'day'))}
```
Comment (majapw) [inline stories/DateRangePicker.js]: We could use the `isSameDay` util method here, but that's somewhat of a nit

**Comment by majapw** [comment]
This looks great. Happy to merge once it is rebased and commits are squashed. I'd like to get this out in this major version bump. :)

**Comment by majapw** [comment]
LGTM! Thanks!

**Pull request merged**